src/wrappers/themis/wasm/src/index.ts

```diff
@@ -28,13 +29,140 @@ export { SecureSession } from "./secure_session";
 export { KeyPair, PrivateKey, PublicKey, SymmetricKey } from "./secure_keygen";
 export { SecureComparator } from "./secure_comparator";
 
+// WebAssembly code is not directly included with compiled JavaScript code.
+// Emscripten generates a stub that will download and compile WebAssembly
+// after this module is loaded. This initialization is necessary before
+// any WasmThemis function can be called.
+//
+// Due to historical reasons, WasmThemis has multiple ways to initialize.
+
+let onRuntimeInitialized: () => void
+
+// The current facility is "async initialize()" function which user code
+// is expected to call, then await the result to resolve before using
+// other WasmThemis functions.
+
+let libthemisInitialized = false;
+
+/**
+ * Initialize WasmThemis.
+ *
+ * This function **must** be called and awaited before using any WasmThemis interfaces.
+ * It will download and compile WebAssembly code of WasmThemis.
+ *
+ * If you are hosting `libthemis.wasm` on a CDN or at a non-standard location,
+ * pass URL to `libthemis.wasm` as an argument.
+ * If URL is omitted, `libthemis.wasm` is expected to be located in the same directory
+ * as the executing script.
+ *
+ * @param wasmPath URL of `libthemis.wasm` to download.
+ *
+ * @throws {ThemisError} is thrown if this function is called more than once,
+ * or if WasmThemis has been already initialized via `initialized`.
+ *
+ * @since WasmThemis 0.14.0
+ */
 export const initialize = async (wasmPath?: string) => {
+  if (libthemisInitialized) {
+    throw new ThemisError(
+      'initialize',
+      ThemisErrorCode.FAIL,
```

Comment:
is explicit error really necessary?

Reply:
> is explicit error really necessary?

Well, one approach is to immediately resolve `initialize()` successfully without actually doing anything if WasmThemis has been already initialized. However, I'd like to send a strong message that you should not call `initialize()` twice. If you do, there's likely some error in your code.

Reply:
> one approach is to immediately resolve initialize() successfully without actually doing anything if WasmThemis has been already initialized

i prefer this approach.

> I'd like to send a strong message that you should not call initialize() twice. If you do, there's likely some error in your code.

but i also agree with you on this.

let's use your version :)

Reply:
Let's think about it like this: we can remove the error signaling and make `initialize()` idempotent in the future, if a genuine use case for that comes along. However, it would be a breaking change to make it an error to call `initialize()` twice, once we slap a note “It's totally okay to call this function multiple time with no consequences!”

Reply:
So I'd rather make it an error now, before the users get to use it, and if they call it twice they get an error saying that it's an error. As opposed to them filing an issue, "I wanted to reinitialize with different settings, the function succeeded, but nothing changed", and us telling them, "well don't do that".